### PR TITLE
Migrate flutter_tester_device to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of https://github.com/flutter/flutter/issues/71511

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
Looks like legit test failure

```
01:28 +2643 ~5: test/general.shard/flutter_tester_device_test.dart: The FLUTTER_TEST environment variable is passed to the test process as null when set to null                                       
Null check operator used on a null value
#0      FlutterTesterTestDevice.start (package:flutter_tools/src/test/flutter_tester_device.dart:145:47)
<asynchronous suspension>
#1      main.<anonymous closure>.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter_tools/test/general.shard/flutter_tester_device_test.dart:181:7)
<asynchronous suspension>
#2      testUsingContext.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter_tools/test/src/context.dart:143:26)
<asynchronous suspension>
#3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#6      testUsingContext.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter_tools/test/src/context.dart:104:5)
<asynchronous suspension>
#7      Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:215:9)
<asynchronous suspension>
#8      Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:213:7)
<asynchronous suspension>
#9      Invoker._waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:257:7)
<asynchronous suspension>

```

File: packages/flutter_tools/lib/src/test/flutter_tester_device.dart [hunk 1]
```diff
@@ -143,38 +142,39 @@ class FlutterTesterTestDevice extends TestDevice {
...
     _pipeStandardStreamsToConsole(
-      process: _process,
+      process: runningProcess,
       reportObservatoryUri: (Uri detectedUri) async {
         assert(!_gotProcessObservatoryUri.isCompleted);
         assert(debuggingOptions.hostVmServicePort == null ||
             debuggingOptions.hostVmServicePort == detectedUri.port);
 
-        Uri forwardingUri;
+        Uri? forwardingUri;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/test/flutter_tester_device.dart:177]:
```suggestion
        final Uri forwardingUri;
```

**Review by christopherfujino — APPROVED** [review]
one nit, otherwise lgtm

**Comment by jmagman** [comment]
> Looks like legit test failure

Yup I hadn't put a reviewer on it yet, I usually don't until the tests are passing.  I will start making ones with failing tests into drafts so they don't show up in your queue.

**Comment by jmagman** [comment]
The problem is that the test is setting a dictionary value to null, which isn't valid for environment variables
https://github.com/flutter/flutter/blob/b2eafd735db50c91aff2b840817410b0b44e6892/packages/flutter_tools/test/general.shard/flutter_tester_device_test.dart#L177-L179

**Pull request closed without merging**